internal/resource/resource.go

```diff
@@ -249,7 +247,7 @@ func NewResource(ctx context.Context, slice *apiv1.ResourceSlice, index int) (*R
 	logger := logr.FromContextOrDiscard(ctx)
 	resource := slice.Spec.Resources[index]
```

Comment:
Unrelated to this PR, but maybe different naming would be beneficial here to differentiate the resource from the slice from our internal representation.